packages/client/src/pages/Signup/validationUtils.ts

```diff
@@ -0,0 +1,104 @@
+import { trim } from '../../utils/validation';
+
+export const validatePassword = (value: string | null) => {
+  const password = trim(value);
+
+  if (password === null || password === '') {
+    return 'Поле обязательно для ввода';
+  }
+
+  if (password.length < 8) {
+    return 'Пароль должен быть не короче 8 символов';
+  }
+
+  if (password.length > 40) {
+    return 'Пароль должен быть не длиннее 40 символов';
+  }
+
+  const regEx =
+    /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,40}$/;
+
+  if (!regEx.test(password)) {
+    return 'Пароль должен состоять из букв латинского алфавита, содержать минимум одну большую и маленькую букву, один спец символ';
+  }
+};
+
+export const validateLogin = (value: string | null) => {
+  const login = trim(value);
+
+  if (login === null || login === '') {
+    return 'Поле обязательно для ввода';
+  }
+
+  if (login.length < 4) {
+    return 'Логин должен быть не короче 4 символов';
+  }
+
+  if (login.length > 20) {
+    return 'Логин должен быть не длиннее 20 символов';
+  }
+
+  const regEx = new RegExp('^[A-Za-z0-9]*$');
+
+  if (!regEx.test(login)) {
+    return 'Логин может состоять только из цифр и букв латинского алфавита';
+  }
+};
+
+export const validateNames = (value: string | null) => {
+  const name = trim(value);
+
+  if (name === null || name === '') {
+    return 'Поле обязательно для ввода';
+  }
```

Comment:
По моему функция trim тут излишняя, так как можно проще:
```suggestion
  if (!value) {
    return 'Поле обязательно для ввода';
  }
  
  const name = value.trim();
```

Reply:
опять-таки неочевидные проблемы с приведениями типов

Reply:
Хотя согласен в случае имени trim это всё же оверкил